src/keeper-helpers.ts

```diff
@@ -0,0 +1,53 @@
+import { ethers } from "ethers";
+import { range } from "lodash";
+
+const MAX_BLOCKS = 2000;
+// exported for test
+export const getPaginatedFromAndTo = (fromBlock: number, toBlock: number) => {
+  const numberOfBlocks = toBlock - fromBlock;
+  if (numberOfBlocks > MAX_BLOCKS) {
```

Comment:
nit: ideally, shouldn't be needed and should return the correct result if numberOfBlocks <= MAX_BLOCKS

Reply:
Hmm we would need some logic for when `numberOfBlocks` is 0, which it is when we're just processing 1 block. 
Can remove the if statement and do this:
```
const numberOfBlocks = toBlock - fromBlock || 1
```
If you think thats clearer?